packages/dds/shared-object-base/src/sharedObject.ts

```diff
@@ -414,7 +421,23 @@ export abstract class SharedObjectCore<TEvent extends ISharedObjectEvents = ISha
     private process(message: ISequencedDocumentMessage, local: boolean, localOpMetadata: unknown) {
         this.verifyNotClosed(); // This will result in container closure.
         this.emit("pre-op", message, local, this);
-        this.processCore(message, local, localOpMetadata);
+
+        // Do not measure for local ops, because callbacks are not invoked for them in this codepath
```

Comment:
I do not think this comment is correct. You are right that in most cases there are no conflicts, or merge policy ensures that merging ack for local op is essentially no-op, and thus events are not fired. But it definitely not true uniformly. 
I'd suggest not to have any special casing here, and instead rely on measuring (and reporting) actual raising of events. If we do not raise an event - there is nothing to report

Reply:
If you want to, you can always add 'local' as a property to event payload, to give a chance for developers to look at various slices of data.

Reply:
> But it definitely not true uniformly.

That's definitely good to know.

The local/remote distinction seemed relevant for the case where we aggregate the data before sending the telemetry event, because then we'd potentially be mixing measurements of dds-processing and dds-processing-plus-callbacks in the same aggregated event and the numbers could skew due to that (plus we'd get very different data depending on how many clients were using the same document, and the corresponding ratio of remote-to-local ops). If we end up sampling, then a `local` property in the payload seems good enough to be able to differentiate those cases in Kusto. I'll address this once the aggregate VS sample topic is finalized.

Reply:
i think regardless of sampling of aggregation we can still support dimensions like this, i think what infra handles the data should support taking a set a property which map to buckets for the data gathering technique, so in this case you could specify local as a top level dimension which the data would be split on.